--- a/PKGBUILD
+++ b/PKGBUILD
@@ -246,7 +246,8 @@
 if [ "$_build_nvidia_open" = "yes" ]; then
     source+=("https://download.nvidia.com/XFree86/${_nv_open_pkg%"-$_nv_ver"}/${_nv_open_pkg}.tar.xz"
              "${_patchsource}/misc/nvidia/0002-fix-dsc-correct-RC-parameter-tables-to-match-VESA-DS.patch"
-             "${_patchsource}/misc/nvidia/0003-fix-dp-add-Bigscreen-Beyond-VR-headset-to-WAR-databa.patch")
+             "${_patchsource}/misc/nvidia/0003-fix-dp-add-Bigscreen-Beyond-VR-headset-to-WAR-databa.patch"
+             "0004-local-hdmi-quirk.patch")
 fi
 
 # Use generated AutoFDO Profile
@@ -311,7 +312,7 @@
         src="${src%.zst}"
         [[ $src = *.patch ]] || continue
         echo "Applying patch $src..."
-        if [[ "$patch" == "${_patchsource}"/misc/nvidia/* ]]; then
+        if [[ "$patch" == "${_patchsource}"/misc/nvidia/* || "$src" == 0004-local-hdmi-quirk.patch ]]; then
             patch -Np1 < "../$src" -d "${srcdir}/${_nv_open_pkg}"
         else
             patch -Np1 < "../$src"
